### DISCUSSION TITLE
Sharing COM Port failed in Hyper-V

### DISCUSSION THREAD

**Discussion by rmshub** [dorssel/usbipd-win#470] [General]
Host: Windows 11 Pro
Guest OS on Hyper-V : Ubuntu 22.04
**Command on Ubuntu**
> usbip list -r 172.23.20.1
Exportable USB devices
======================
 - 172.23.20.1
        1-2: Silicon Labs : CP210x UART Bridge (10c4:ea60)
           : USB\VID_10C4&PID_EA60\DCC11553D40EEC11BB9EB9EB7A109228
           : (Defined at Interface level) (00/00/00)
           :  0 - Vendor Specific Class / unknown subclass / unknown protocol (ff/00/00)

> sudo usbip attach -r 172.23.20.1 -b 1-2
libusbip: error: udev_device_new_from_subsystem_sysname failed
usbip: error: open vhci_driver

**Usbipd log on windows**
dbug: Microsoft.Extensions.Hosting.Internal.Host[1]
      Hosting starting
dbug: Usbipd.PcapNg[1000]
      usbipd:PcapNg:Path = ''
dbug: Usbipd.Server[1000]
      2.4.1+1.Branch.master.Sha.90acf9456020ca8c6310ca62a71ee23cb6ca34ad
dbug: Usbipd.Server[1000]
      usbipd:TcpKeepAliveInterval = 500 ms
dbug: Usbipd.Server[1000]
      usbipd:TcpKeepAliveTime = 10000 ms
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\Users\
dbug: Microsoft.Extensions.Hosting.Internal.Host[2]
      Hosting started
dbug: Usbipd.Server[1000]
      new connection from 172.23.20.65
dbug: Usbipd.ConnectedClient[1000]
      Received opcode: OP_REQ_IMPORT
info: Usbipd.ConnectedClient[1]
      Client 172.23.20.65 claimed device at 1-2 (USB\VID_10C4&PID_EA60\DCC11553D40EEC11BB9EB9EB7A109228).
info: Usbipd.ConnectedClient[2]
      Client 172.23.20.65 released device at 1-2 (USB\VID_10C4&PID_EA60\DCC11553D40EEC11BB9EB9EB7A109228).
dbug: Usbipd.ConnectedClient[1000]
      Unbind or unplug while attached

Could you please help me to resolve this issue ?

**Comment by dorssel** [comment]
You need to load the USBIP driver on Linux first, before attaching:
```
sudo modprobe vhci-hcd
sudo usbip attach -r 172.23.20.1 -b 1-2
```

**Reply by rmshub** [reply]
Thanks for quick update

Got this error "modprobe: FATAL: "Module vhci-hcd not found" in directory /lib/modules/5.15.0-1022-azure"

**Reply by dorssel** [reply]
You need the generic Linux image, not the Azure one.

```
sudo apt install linux-image-generic
```

**Comment by rmshub** [comment]
After changing kernel image to generic image above commands are working properly

Refer https://askubuntu.com/questions/82140/how-can-i-boot-with-an-older-kernel-version to switch different kernel version
